src/services/registrationData.ts

```diff
@@ -103,22 +103,173 @@ export async function upsertRegistrationData({
 }
 
 /**
- * Function: updateQuestionOptions
- * Updates or inserts registration data into the question options table.
- * @param dbPool - The database pool instance of type `PostgresJsDatabase<typeof db>`.
- * @param registrationData - An array of objects representing registration data.
- *   Each object should have the properties:
- *   - id: The unique identifier for the registration data.
- *   - registrationFieldId: The identifier for the registration field associated with the data.
- *   - value: The value of the registration data.
- * @returns A Promise that resolves once the update/insert operation is completed.
+ * Fetches registration fields from the database.
+ * @param dbPool - The database pool instance.
+ * @param registrationFieldIds - Array of registration field IDs.
+ * @returns An array of registration fields.
  */
-export async function updateQuestionOptions(
+async function fetchRegistrationFields(
+  dbPool: PostgresJsDatabase<typeof db>,
+  registrationFieldIds: string[],
+): Promise<
+  {
+    registrationFieldId: string;
+    questionId: string;
+    questionOptionType: string;
+  }[]
+> {
+  // Fetch registration fields from the database based on the provided IDs
+  const registrationFields = await dbPool.execute<{
+    registrationFieldId: string;
+    questionId: string;
+    questionOptionType: string;
+  }>(
+    sql.raw(`
+        SELECT id AS "registrationFieldId", question_id AS "questionId", question_option_type AS "questionOptionType"
+        FROM registration_fields
+        WHERE question_id IS NOT NULL
+        AND question_option_type IN ('TITLE', 'SUBTITLE')
+        AND id IN (${registrationFieldIds.map((id) => `'${id}'`).join(', ')})
+        `),
+  );
+
+  return registrationFields;
+}
+
+/**
+ * Filters registration data based on the available registration fields.
+ * @param registrationData - An array of registration data.
+ * @param registrationFields - An array of registration fields.
+ * @returns Filtered registration data.
+ */
+function filterRegistrationData(
+  registrationData: {
+    id: string;
+    registrationFieldId: string;
+    registrationId: string;
+    value: string;
+  }[],
+  registrationFields: {
+    registrationFieldId: string;
+    questionId: string;
+    questionOptionType: string;
+  }[],
+): {
+  id: string;
+  registrationFieldId: string;
+  registrationId: string;
+  value: string;
+}[] {
+  // Filter registration data to include only entries that require updating
+  const filteredRegistrationData = registrationData.filter((data) =>
+    registrationFields.some((field) => field.registrationFieldId === data.registrationFieldId),
+  );
+
+  return filteredRegistrationData;
+}
+
+/**
+ * Maps filtered registration data to a combined format.
+ * @param filteredRegistrationData - Filtered registration data.
+ * @returns Combined registration data.
+ */
+function mapToCombinedData(
```

Comment:
would rename this function to something that describes it more, like: `transformRegistrationDataToCombinedFormat`